plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,467 @@
+
+  @classmethod
+  def GetFormatSpecification(cls):
+    """Retrieves the format specification.
+
+    Returns:
+      FormatSpecification: format specification.
+    """
+    format_specification = specification.FormatSpecification(cls.NAME)
+    format_specification.AddNewSignature(cls.FORMAT_SIGNATURE, offset=0)
+    return format_specification
+
+  def _GetFileEntryInSameDirectory(
+      self, sibling_entry, target_filename, file_system, parser_mediator):
+    """Gets a file entry from the same directory as another file.
+
+    Args:
+      target_filename (str): the name of the file to open.
+      sibling_entry (dfvfs.FileEntry): the sibling file entry.
+      file_system (dfvfs.FileSystem): the file system containing the OneDrive
+        logs.
+      parser_mediator (ParserMediator): mediates interactions between parsers
+        and other components, such as storage and dfVFS.
+
+    Returns:
+      dfvfs.FileEntry: the target file entry.  None when the target file is not
+          found or able to be opened.
+    """
+    path_segments = file_system.SplitPath(sibling_entry.path_spec.location)
+    path_segments.pop()
+    path_segments.append(target_filename)
+
+    location = file_system.JoinPath(path_segments)
+    kwargs = {
+        'location': location,
+        'parent': None}
+
+    filename_path_spec = path_spec_factory.Factory.NewPathSpec(
+        sibling_entry.path_spec.TYPE_INDICATOR, **kwargs)
+
+    try:
+      filename_file_entry = path_spec_resolver.Resolver.OpenFileEntry(
+          filename_path_spec)
+    except RuntimeError as exception:
+      parser_mediator.ProduceExractionWarning(
+          'Unable to open filename: {0:s} with error: {1!s}'.format(
+              location, exception))
+      return None
+
+    return filename_file_entry
+
+  def _ParseObfuscatedStringMap(
+      self, file_entry, file_system, parser_mediator):
+    """Parses the ObfuscatedStringMap.txt file.
+
+    Args:
+      file_entry (dfvfs.FileEntry): the OneDrive Log file entry.
+      file_system (dfvfs.FileSystem): the file system containing the OneDrive
+        logs.
+      parser_mediator  (ParserMediator): mediates interactions between parsers
+        and other components, such as storage and dfVFS.
+
+    Returns:
+      dict[str, str]: a mapping of an obfuscated string value to its true string
+        value.  None if the ObfuscatedSTringMap.txt file could not be found.
+    """
+    obfuscated_string_map_entry = self._GetFileEntryInSameDirectory(
+        file_entry, self._OBFUSCATED_STRING_MAP_FILE, file_system,
+        parser_mediator)
+
+    if not obfuscated_string_map_entry:
+      return None
+
+    obfuscated_string_map_file = obfuscated_string_map_entry.GetFileObject()
+
+    obfuscated_string_map = {}
+    with text_file.TextFile(obfuscated_string_map_file,
+        encoding='utf-16-le') as obfuscated_text_file:
+      for line in obfuscated_text_file.readlines():
+        cleaned_line = line.rstrip().split('\t')
```

Comment:
so cleaned_line is not a single string but the split values, please rename this variable to closer represents its purpose

Reply:
Done